### PR TITLE
Kaitie/aitt add data to cw experience

### PR DESCRIPTION
**Pull request by kobryan0619**
This is intended to connect the relevant lesson data to the Rubric Edit and New Rubric pages.   Since this is the first time I have worked so much with ruby and haml in our code base, I am looking for feedback before I continue to add the data pieces for the rubrics.  

After this PR we should:

- Start and finish the "save rubric" functionality
- Finish pulling in the data from an existing rubric.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/lib/levelbuilder/lesson-editor/LessonEditor.jsx
Comment (kobryan0619) [inline apps/src/lib/levelbuilder/lesson-editor/LessonEditor.jsx]: I think this works but since there are no exisiting lessons with rubrics, I could only test it by passing true and false in for this prop.

File: apps/src/lib/levelbuilder/rubrics/RubricsContainer.jsx
Comment (kobryan0619) [inline apps/src/lib/levelbuilder/rubrics/RubricsContainer.jsx]: I know that I need a "shape" here, but none of the existing "shapes" seemed to match what I had.  I am planning to ask about this in my next pairing session, but open to feedback here too

File: apps/src/sites/studio/pages/rubrics/edit.js
Comment (kobryan0619) [inline apps/src/sites/studio/pages/rubrics/edit.js:17]: I am hoping to add the hasRubric prop here when we have rubric data ready to go.

File: apps/src/sites/studio/pages/rubrics/new.js
Comment (kobryan0619) [inline apps/src/sites/studio/pages/rubrics/new.js:16]: Again, hoping to add the hasRubric prop here once we have rubrics :)

File: apps/test/unit/lib/levelbuilder/lesson-editor/LessonEditorTest.js
Comment (kobryan0619) [inline apps/test/unit/lib/levelbuilder/lesson-editor/LessonEditorTest.js:390]: Wondering: Should I also check to see that the hrefs are accurate here?

File: apps/src/lib/levelbuilder/lesson-editor/LessonEditor.jsx
Comment (bethanyaconnor) [inline apps/src/lib/levelbuilder/lesson-editor/LessonEditor.jsx]: I think you'll need to pipe the rubric id down here. So, maybe instead of `hasRubric`, you can use a prop `rubricId` -- if it's null, there's no rubric, if it has a value, there is one

File: apps/src/lib/levelbuilder/rubrics/RubricsContainer.jsx
Comment (bethanyaconnor) [inline apps/src/lib/levelbuilder/rubrics/RubricsContainer.jsx]: I think this should work for standalone units? And standalone lessons don't exist right now so no need to worry about them!

File: apps/src/lib/levelbuilder/rubrics/RubricsContainer.jsx
Comment (bethanyaconnor) [inline apps/src/lib/levelbuilder/rubrics/RubricsContainer.jsx]: You may need to update these to be `level.id` for the value and `level.name` for the display

File: apps/src/sites/studio/pages/rubrics/edit.js
Comment (bethanyaconnor) [reply apps/src/sites/studio/pages/rubrics/edit.js:17]: Do you need the `hasRubric` prop for the rubric editor?

File: apps/test/unit/lib/levelbuilder/lesson-editor/LessonEditorTest.js
Comment (bethanyaconnor) [reply apps/test/unit/lib/levelbuilder/lesson-editor/LessonEditorTest.js:390]: That sounds like a great idea! You should be able to do something like `wrapper.find('.btn.add-rubric').props().href` and check equality

File: dashboard/app/controllers/rubrics_controller.rb
Comment (bethanyaconnor) [inline dashboard/app/controllers/rubrics_controller.rb]: The ID supplied in the URL for the rubric editor will be the rubric id, not the lesson id. If you need the lesson, you can always get it from the rubric

File: dashboard/app/models/lesson.rb
Comment (bethanyaconnor) [inline dashboard/app/models/lesson.rb]: nit: use `!!rubric` or `rubric.present?`

File: dashboard/app/models/lesson.rb
Comment (bethanyaconnor) [inline dashboard/app/models/lesson.rb]: do you need the whole unit for this page or just the name?

**Comment by bethanyaconnor** [comment]
Great start! I think I ended up adding a lot of comments but most are nits or ideas for improvements! It sounds like you're still working on finishing this up but want to make sure getting the rubric data down to the edit page and saving (both on the new and edit pages) are both on your list?

File: dashboard/app/models/lesson.rb
Comment (kobryan0619) [reply dashboard/app/models/lesson.rb]: Just the name is needed.  I made this change.

File: dashboard/app/controllers/rubrics_controller.rb
Comment (kobryan0619) [reply dashboard/app/controllers/rubrics_controller.rb]: I made this change but still am a bit uncertain how to test it without any existing data.

File: apps/src/lib/levelbuilder/rubrics/RubricsContainer.jsx
Comment (bethanyaconnor) [reply apps/src/lib/levelbuilder/rubrics/RubricsContainer.jsx]: I've got a shape for rubrics that I'm working on that might work here. As for lesson, you might just be able to do an inline shape as you only need a name and id. Something like:

```levels: PropTypes.arrayOf(PropType.shape({id: PropTypes.number, name: PropTypes.string}))```

File: dashboard/app/models/lesson.rb
Comment (bethanyaconnor) [inline dashboard/app/models/lesson.rb]: nit: we don't need the `localized_name` for the editor; `name` is fine (editing can only happen in English)

**Review by bethanyaconnor — APPROVED** [review]
Looks good -- there are a couple of comments (a nit and a little bit of additional testing) that I'd like to see addressed before merge but I'm approving now as they should be pretty minor :) 

Can you update the PR description as well? I think there's at least one more PR before we can consider the editor done.

File: dashboard/app/controllers/rubrics_controller.rb
Comment (kobryan0619) [inline dashboard/app/controllers/rubrics_controller.rb]: I was trying to find an efficient way to pass around this data.  For a while I had a query param as well in the URL which also worked, but this seemed more efficient.  That being said, I looked in our code base for anywhere else that did anything similar to this and I couldn't find anything which gave me pause.

Comment (bethanyaconnor) [reply dashboard/app/controllers/rubrics_controller.rb]: What data do you mean, sorry? The lesson? You can just do `@lesson = @rubric.lesson` to get that (it's pretty much the same as what you wrote but more "rails-y")

**Review by bethanyaconnor — APPROVED** [review]
Added one comment on the new changes but otherwise looks good!

**Pull request merged**
